src/lib.rs

```diff
@@ -1,14 +1,38 @@
-pub fn add(left: usize, right: usize) -> usize {
-    left + right
-}
+#![feature(arbitrary_enum_discriminant)]
 
-#[cfg(test)]
-mod tests {
-    use super::*;
+pub mod incoming;
 
-    #[test]
-    fn it_works() {
-        let result = add(2, 2);
-        assert_eq!(result, 4);
-    }
-}
+/// A simple type alias for a result whose error case is an IRC error.
+pub type IrcResult<T> = Result<T, IrcError>;
+
+#[repr(u16)]
```

Comment:
Add a comment why we are representing this u16

Reply:
Reasons will change depending on the decision to use the arbitrary discriminants feature.